<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Erie Data: Unconstrained Ordination</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="060_Erie_Data_Unconstrained_Ordination_files/libs/clipboard/clipboard.min.js"></script>
<script src="060_Erie_Data_Unconstrained_Ordination_files/libs/quarto-html/quarto.js"></script>
<script src="060_Erie_Data_Unconstrained_Ordination_files/libs/quarto-html/popper.min.js"></script>
<script src="060_Erie_Data_Unconstrained_Ordination_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="060_Erie_Data_Unconstrained_Ordination_files/libs/quarto-html/anchor.min.js"></script>
<link href="060_Erie_Data_Unconstrained_Ordination_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="060_Erie_Data_Unconstrained_Ordination_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="060_Erie_Data_Unconstrained_Ordination_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="060_Erie_Data_Unconstrained_Ordination_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="060_Erie_Data_Unconstrained_Ordination_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Erie Data: Unconstrained Ordination</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<section id="introduction" class="level2">
<h2 class="anchored" data-anchor-id="introduction">Introduction</h2>
<p>Multivariable data with a large number of variables can be difficult to interpret. In order to understand relationships in the data we need to summarize the variability in a way that reduces the dimensionality of the data. <em>Ordination</em> attempts to order the samples/observations in a data set along a small number of axes so that similar samples are close to each other and dissimilar samples are far from each other.</p>
<p>There are several different ordination methods that have been applied to ecological data in the past. Ordination methods fall into two categories: <em>constrained ordination</em> and <em>unconstrained ordination</em>. Both groups of methods start with an abundance matrix with samples along one dimension (e.g., rows) and species (or other taxonomic classification) along the the other (e.g., columns). Two samples are similar if they have similar species compositions. Unconstrained ordination uses information based on species composition to position samples along the ordination axes. Constrained ordination also uses environmental data, and attempts to reveal whether species composition is associated with the environmental variables.</p>
<p>In this tutorial we will explore the Lake Erie bacterial community data using two unconstrained ordination methods, Principal Coordinate Analysis (PCoA) and Nonmetric Multidimensional Scaling (NMDS). Both of these methods are considered distance-based methods because they are both based on a distance matrix that is computed from the species abundance matrix. The distance between two samples can be calculated in a variety of ways. For example, the Euclidean distance can be computed between the abundance vectors for two samples. The distance matrix has the samples along both dimensions (rows and columns), so that the element in the <span class="math inline">\(i\)</span>th row and the <span class="math inline">\(j\)</span>th column is the distance between sample <span class="math inline">\(i\)</span> and sample <span class="math inline">\(j\)</span>.</p>
<p>The number of dimensions of the abundance data is equal to the number of species. For ease of visualization we would like to reduce the dimensionality of the data so that it can be represented in two or (less often) three dimensions. However, we would like to do this in a way that preserves as much of the original patterns of variation as possible. PCoA rotates the higher dimensional data cloud and projects it into lower dimension in such a way as to maximize the linear correlations between the distances in the lower dimensional representation and the distances in the original data. NMDS positions the samples in the lower dimensional space while attempting to maximize the agreement between the rank order of distances in the lower dimensional representation and the rank order of distances in the original data. The <em>stress</em> measures the disagreement between the two rank orders.</p>
<p>There are a variety of ways to perform ordination analyses in R. Many of the ordination techniques used in ecology have been implemented in the package <em>vegan</em>. We will use this package for caclulating distance matrices and for NMDS. We will use the <em>ape</em> package for PCoA.</p>
</section>
<section id="your-quarto-document" class="level2">
<h2 class="anchored" data-anchor-id="your-quarto-document">Your Quarto document</h2>
<p>Create a new Quarto document for this tutorial, give it an informative name, and save it in the <em>code</em> subdirectory of your <em>watershedErie</em> directory. As you work through this tutorial, transcribe the code into your own document and run it. Make sure you also write about what you are doing so that you can understand your work when you read it later.</p>
</section>
<section id="setup-and-reading-in-the-clean-data" class="level2">
<h2 class="anchored" data-anchor-id="setup-and-reading-in-the-clean-data">Setup and reading in the clean data</h2>
<p>Let’s start by cleaning up our work space and loading the required packages. If you have not installed the <em>vegan</em> and <em>ape</em> packages, do so now.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list =</span> <span class="fu">ls</span>())</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">gc</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>          used (Mb) gc trigger (Mb) limit (Mb) max used (Mb)
Ncells  579876 31.0    1322067 70.7         NA   669442 35.8
Vcells 1067917  8.2    8388608 64.0     102400  1851737 14.2</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.3     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.3     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(vegan)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: permute
Loading required package: lattice
This is vegan 2.6-4</code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ape)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'ape'

The following object is masked from 'package:dplyr':

    where</code></pre>
</div>
</div>
</section>
<section id="import-the-data" class="level2">
<h2 class="anchored" data-anchor-id="import-the-data">Import the data</h2>
<p>Let’s load the filtered data. Note that this is not the aggregated data set that we used in the last tutorial.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>erie <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"clean_data/erie.rds"</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>erie_tax <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"clean_data/erie_tax.rds"</span>)</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>erie_meta <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"clean_data/erie_meta.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="remove-samples-with-a-small-number-of-reads-10k" class="level2">
<h2 class="anchored" data-anchor-id="remove-samples-with-a-small-number-of-reads-10k">Remove samples with a small number of reads (&lt;10k)</h2>
<p>We will remove any samples with a sequencing depth of less than 10,000 reads. Since we added the sequencing depth to the metadata this is easy to do.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>erie_meta <span class="ot">&lt;-</span> erie_meta <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sequencingDepth <span class="sc">&gt;=</span> <span class="dv">10000</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>erie_samples <span class="ot">&lt;-</span> erie_meta <span class="sc">%&gt;%</span> <span class="fu">pull</span>(SampleID)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>erie <span class="ot">&lt;-</span> erie <span class="sc">%&gt;%</span> <span class="fu">filter</span>(SampleID <span class="sc">%&gt;%</span> <span class="fu">is.element</span>(erie_samples))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="computing-relative-abundances-and-filtering-the-data" class="level2">
<h2 class="anchored" data-anchor-id="computing-relative-abundances-and-filtering-the-data">Computing Relative Abundances and Filtering the Data</h2>
<p>We will normalize each sample by dividing by by the total number of sequencing reads. This is similar to how we computed relative abundances in Tutorial 5. This time, however, we are not aggregating the data first, so the results will be the relative abundances for the OTUs in each sample rather than the relative abundances for the phyla.</p>
<p>Start by lengthening the data and grouping by SampleID.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>erie <span class="ot">&lt;-</span> erie <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">starts_with</span>(<span class="st">"Otu"</span>),</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">"OTU"</span>,</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">"Abundance"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(SampleID)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Scale the data to get relative abundances, and remove rows with relative abundance below 0.005%.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>erie <span class="ot">&lt;-</span> erie <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Rel_Abundance =</span> Abundance<span class="sc">/</span><span class="fu">sum</span>(Abundance)) <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Rel_Abundance <span class="sc">&gt;</span> <span class="fl">5e-5</span>)</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>erie</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 27,902 × 6
# Groups:   SampleID [56]
   label SampleID  numOtus OTU      Abundance Rel_Abundance
   &lt;dbl&gt; &lt;chr&gt;       &lt;int&gt; &lt;chr&gt;        &lt;int&gt;         &lt;dbl&gt;
 1  0.03 E0010_CNA   19656 Otu00002      1271      0.0445  
 2  0.03 E0010_CNA   19656 Otu00004      3264      0.114   
 3  0.03 E0010_CNA   19656 Otu00005         3      0.000105
 4  0.03 E0010_CNA   19656 Otu00006        41      0.00143 
 5  0.03 E0010_CNA   19656 Otu00007       110      0.00385 
 6  0.03 E0010_CNA   19656 Otu00008      1351      0.0473  
 7  0.03 E0010_CNA   19656 Otu00010      1956      0.0684  
 8  0.03 E0010_CNA   19656 Otu00011      3162      0.111   
 9  0.03 E0010_CNA   19656 Otu00012        48      0.00168 
10  0.03 E0010_CNA   19656 Otu00013       139      0.00486 
# ℹ 27,892 more rows</code></pre>
</div>
</div>
<p>We probably dropped some OTUs from the data altogether. Even though we won’t use the tax table here, let’s filter it down to the OTUs we kept. We’ll use this opportunity to see how many OTUs we have lost by counting rows in <em>erie_tax</em> tibble before and after dropping the OTUs.</p>
<p>We will use a filtering join (<em>semi_join</em>) to filter the OTUs out of <em>erie_tax</em> that are no longer in <em>erie</em>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(erie_tax)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 7522</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>erie_tax <span class="ot">&lt;-</span> erie_tax <span class="sc">%&gt;%</span> <span class="fu">semi_join</span>(erie, <span class="at">by =</span> <span class="st">"OTU"</span>) <span class="co">#filtering join</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(erie_tax)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 4862</code></pre>
</div>
</div>
<p>Now we pivot back to wide format (one row per sample). The command will insert zeros when there are missing values (OTUs that were filtered out for some samples but not others). The OTUs that we dropped from all samples will not be in the resulting data frame. We drop the abundances and proceed with only the relative abundances.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>erie <span class="ot">&lt;-</span> erie <span class="sc">%&gt;%</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  ungroup <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Abundance)) <span class="sc">%&gt;%</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> OTU, <span class="at">values_from =</span> Rel_Abundance, <span class="at">values_fill =</span> <span class="dv">0</span>)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>erie</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 56 × 4,865
   label SampleID  numOtus Otu00002 Otu00004 Otu00005 Otu00006 Otu00007 Otu00008
   &lt;dbl&gt; &lt;chr&gt;       &lt;int&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;    &lt;dbl&gt;
 1  0.03 E0010_CNA   19656   0.0445   0.114  0.000105 0.00143   0.00385   0.0473
 2  0.03 E0012_CNA   19656   0.0280   0.0948 0.000375 0.00296   0.0325    0.0476
 3  0.03 E0014_CNA   19656   0.0578   0.115  0.000199 0.000598  0.0113    0.0315
 4  0.03 E0016_CNA   19656   0.0525   0.0877 0        0.00943   0.0427    0.0328
 5  0.03 E0018_CNA   19656   0.0454   0.0942 0        0.00925   0.0596    0.0465
 6  0.03 E0020_CNA   19656   0.0439   0.101  0.00411  0.0154    0.0407    0.0518
 7  0.03 E0022_CNA   19656   0.0455   0.104  0.00204  0.0216    0.0344    0.0482
 8  0.03 E0024_CNA   19656   0.0401   0.0772 0.000243 0.0224    0.0318    0.0445
 9  0.03 E0026_CNA   19656   0.0449   0.0905 0.00258  0.0280    0.0145    0.0379
10  0.03 E0028_CNA   19656   0.0340   0.0766 0.0164   0.0696    0.0335    0.0286
# ℹ 46 more rows
# ℹ 4,856 more variables: Otu00010 &lt;dbl&gt;, Otu00011 &lt;dbl&gt;, Otu00012 &lt;dbl&gt;,
#   Otu00013 &lt;dbl&gt;, Otu00015 &lt;dbl&gt;, Otu00016 &lt;dbl&gt;, Otu00017 &lt;dbl&gt;,
#   Otu00018 &lt;dbl&gt;, Otu00019 &lt;dbl&gt;, Otu00023 &lt;dbl&gt;, Otu00025 &lt;dbl&gt;,
#   Otu00026 &lt;dbl&gt;, Otu00027 &lt;dbl&gt;, Otu00028 &lt;dbl&gt;, Otu00029 &lt;dbl&gt;,
#   Otu00031 &lt;dbl&gt;, Otu00033 &lt;dbl&gt;, Otu00034 &lt;dbl&gt;, Otu00035 &lt;dbl&gt;,
#   Otu00036 &lt;dbl&gt;, Otu00038 &lt;dbl&gt;, Otu00039 &lt;dbl&gt;, Otu00040 &lt;dbl&gt;, …</code></pre>
</div>
</div>
</section>
<section id="community-matrix" class="level2">
<h2 class="anchored" data-anchor-id="community-matrix">Community matrix</h2>
<p>To do the ordination, we need to create a <em>community matrix</em> with rows corresponding to samples and columns corresponding to OTUs. Our <em>erie</em> tibble already has this shape, but it has some extra columns (label and numOtus), and it is a <em>tibble</em> object instead of a matrix. A matrix is a simpler data structure. There are several ways in which it is different from a <em>tibble</em>. One is that all of the data must be of the same type in a matrix. We can give the matrix row and column names, so we name the rows using the sample IDs, and we name the columns according to the OTUs. All of the entries in the matrix are just relative abundances (numbers). The row and column names help us to associate the numbers with the correnct samples and OTUs.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>erie_rownames <span class="ot">&lt;-</span> erie <span class="sc">%&gt;%</span> <span class="fu">pull</span>(SampleID)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>erie_matrix <span class="ot">&lt;-</span> erie <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">"Otu"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.matrix</span>() <span class="sc">%&gt;%</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">rownames&lt;-</span><span class="st">`</span>(erie_rownames)</span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>erie_matrix <span class="sc">%&gt;%</span> dim</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]   56 4862</code></pre>
</div>
</div>
</section>
<section id="pcoa" class="level2">
<h2 class="anchored" data-anchor-id="pcoa">PCOA</h2>
<p>We can use the ape::pcoa function to perform pcoa, but we need to compute a dissimilarity/distance matrix first. We can use the <em>vegdist</em> command in the <em>vegan</em> package to do that. The <em>vegdist</em> command takes a community matrix and a method as inputs. We will use the “bray” method, which computes Bray-Curtis dissimilarities.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>erie_dist <span class="ot">&lt;-</span> erie_matrix <span class="sc">%&gt;%</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">vegdist</span>(<span class="at">method =</span> <span class="st">"bray"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let’s save the distance matrix in case we need to use it again.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>erie_dist <span class="sc">%&gt;%</span> <span class="fu">saveRDS</span>(<span class="st">"clean_data/erie_dist_matrix_bray.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, performing the PCoA is straightforward. We just use the <em>pcoa</em> command from the <em>ape</em> package.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>erie_pcoa <span class="ot">&lt;-</span> erie_dist <span class="sc">%&gt;%</span> <span class="fu">pcoa</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We will plot our data using the first two principal coordinates. The following code creates a tibble with the coordinates of each sample along the two first two PCoA axes.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>erie_pcoa_tib <span class="ot">&lt;-</span> erie_pcoa <span class="sc">%&gt;%</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pluck</span>(<span class="st">"vectors"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  .[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>] <span class="sc">%&gt;%</span> </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>( <span class="at">pcoa.1 =</span> Axis<span class="fl">.1</span>, <span class="at">pcoa.2 =</span> Axis<span class="fl">.2</span>)</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>erie_pcoa_tib</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 56 × 3
   SampleID  pcoa.1   pcoa.2
   &lt;chr&gt;      &lt;dbl&gt;    &lt;dbl&gt;
 1 E0010_CNA -0.336  0.00137
 2 E0012_CNA -0.276 -0.0670 
 3 E0014_CNA -0.334 -0.00544
 4 E0016_CNA -0.271 -0.108  
 5 E0018_CNA -0.247 -0.133  
 6 E0020_CNA -0.175 -0.165  
 7 E0022_CNA -0.236 -0.109  
 8 E0024_CNA -0.169 -0.180  
 9 E0026_CNA -0.191 -0.136  
10 E0028_CNA -0.133 -0.135  
# ℹ 46 more rows</code></pre>
</div>
</div>
<p>We will join this tibble with the Erie metadata, so that we can use variables in the metadata in the PCoA plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>erie_meta <span class="ot">&lt;-</span> erie_meta <span class="sc">%&gt;%</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(erie_pcoa_tib)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(SampleID)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>erie_meta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 56 × 35
   SampleID  Date       Station   Shore Fraction Type  Samplenum Date_year Month
   &lt;chr&gt;     &lt;date&gt;     &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;
 1 E0010_CNA 2014-06-16 nearshor… near… CNA      samp… E0010     6/16/14   June 
 2 E0012_CNA 2014-06-16 nearshor… near… CNA      samp… E0012     6/16/14   June 
 3 E0014_CNA 2014-06-16 offshore  offs… CNA      samp… E0014     6/16/14   June 
 4 E0016_CNA 2014-06-30 nearshor… near… CNA      samp… E0016     6/30/14   June 
 5 E0018_CNA 2014-06-30 nearshor… near… CNA      samp… E0018     6/30/14   June 
 6 E0020_CNA 2014-06-30 offshore  offs… CNA      samp… E0020     6/30/14   June 
 7 E0022_CNA 2014-07-08 nearshor… near… CNA      samp… E0022     7/8/14    July 
 8 E0024_CNA 2014-07-08 nearshor… near… CNA      samp… E0024     7/8/14    July 
 9 E0026_CNA 2014-07-08 offshore  offs… CNA      samp… E0026     7/8/14    July 
10 E0028_CNA 2014-07-14 nearshor… near… CNA      samp… E0028     7/14/14   July 
# ℹ 46 more rows
# ℹ 26 more variables: Days &lt;dbl&gt;, pH &lt;dbl&gt;, SpCond &lt;dbl&gt;, StationDepth &lt;dbl&gt;,
#   Secchi &lt;dbl&gt;, Temp &lt;dbl&gt;, Turbidity &lt;dbl&gt;, ParMC &lt;dbl&gt;, DissMC &lt;dbl&gt;,
#   Phycocyanin &lt;dbl&gt;, Chla &lt;dbl&gt;, DOC &lt;dbl&gt;, SRP &lt;dbl&gt;, TP &lt;dbl&gt;, TDP &lt;dbl&gt;,
#   POC &lt;dbl&gt;, PON &lt;dbl&gt;, PP &lt;dbl&gt;, N.P &lt;dbl&gt;, Nitrate &lt;dbl&gt;, Ammonia &lt;dbl&gt;,
#   H2O2 &lt;dbl&gt;, CDOM &lt;dbl&gt;, sequencingDepth &lt;int&gt;, pcoa.1 &lt;dbl&gt;, pcoa.2 &lt;dbl&gt;</code></pre>
</div>
</div>
</section>
<section id="nmds" class="level2">
<h2 class="anchored" data-anchor-id="nmds">NMDS</h2>
<p>We will use the <em>metaMDS</em> function in <em>vegan</em> to perform the NMDS ordination. We can use the same dissimilarity matrix we computed earlier.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">8675309</span>) <span class="co">#seed the rng so that we all get the same results</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>erie_nmds <span class="ot">&lt;-</span> erie_dist <span class="sc">%&gt;%</span> <span class="fu">metaMDS</span>(<span class="at">trymax =</span> <span class="dv">200</span>, <span class="at">sfgrmin =</span> <span class="fl">1e-9</span>, <span class="at">sratmax =</span> <span class="fl">0.999999999</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Run 0 stress 0.1288091 
Run 1 stress 0.1310707 
Run 2 stress 0.1340905 
Run 3 stress 0.129845 
Run 4 stress 0.1507764 
Run 5 stress 0.1402772 
Run 6 stress 0.1297044 
Run 7 stress 0.1460872 
Run 8 stress 0.1483734 
Run 9 stress 0.160422 
Run 10 stress 0.1521334 
Run 11 stress 0.1377881 
Run 12 stress 0.1302375 
Run 13 stress 0.1507874 
Run 14 stress 0.160927 
Run 15 stress 0.1449993 
Run 16 stress 0.1297434 
Run 17 stress 0.1360442 
Run 18 stress 0.1584935 
Run 19 stress 0.1301502 
Run 20 stress 0.1290188 
... Procrustes: rmse 0.01966698  max resid 0.1075564 
Run 21 stress 0.1558269 
Run 22 stress 0.1275425 
... New best solution
... Procrustes: rmse 0.03374706  max resid 0.1920336 
Run 23 stress 0.1433967 
Run 24 stress 0.1364208 
Run 25 stress 0.1310707 
Run 26 stress 0.1295761 
Run 27 stress 0.130164 
Run 28 stress 0.1297434 
Run 29 stress 0.1359667 
Run 30 stress 0.1297956 
Run 31 stress 0.1377881 
Run 32 stress 0.1302375 
Run 33 stress 0.1322382 
Run 34 stress 0.1278569 
... Procrustes: rmse 0.01864103  max resid 0.1002752 
Run 35 stress 0.1598183 
Run 36 stress 0.1305527 
Run 37 stress 0.1367497 
Run 38 stress 0.1362864 
Run 39 stress 0.1340919 
Run 40 stress 0.1342174 
Run 41 stress 0.1376068 
Run 42 stress 0.1434103 
Run 43 stress 0.1515569 
Run 44 stress 0.154211 
Run 45 stress 0.131142 
Run 46 stress 0.1275425 
... New best solution
... Procrustes: rmse 8.181112e-08  max resid 3.464119e-07 
... Similar to previous best
*** Best solution repeated 1 times</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>erie_nmds</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
metaMDS(comm = ., trymax = 200, sfgrmin = 1e-09, sratmax = 0.999999999) 

global Multidimensional Scaling using monoMDS

Data:     . 
Distance: bray 

Dimensions: 2 
Stress:     0.1275425 
Stress type 1, weak ties
Best solution was repeated 1 time in 46 tries
The best solution was from try 46 (random start)
Scaling: centring, PC rotation, halfchange scaling 
Species: scores missing</code></pre>
</div>
</div>
<p>The stress is 0.128, which is considered acceptable since it is less than 0.2.</p>
<p>Let’s create a tibble with the NMDS coordinates.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>erie_nmds_tib <span class="ot">&lt;-</span> erie_nmds <span class="sc">%&gt;%</span> </span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pluck</span>(<span class="st">"points"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as_tibble</span>(<span class="at">rownames =</span> <span class="st">"SampleID"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">nmds.1 =</span> MDS1, <span class="at">nmds.2 =</span> MDS2)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>erie_nmds_tib</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 56 × 3
   SampleID  nmds.1  nmds.2
   &lt;chr&gt;      &lt;dbl&gt;   &lt;dbl&gt;
 1 E0010_CNA -0.664 -0.113 
 2 E0012_CNA -0.568 -0.201 
 3 E0014_CNA -0.558 -0.0642
 4 E0016_CNA -0.495 -0.193 
 5 E0018_CNA -0.414 -0.190 
 6 E0020_CNA -0.337 -0.295 
 7 E0022_CNA -0.380 -0.132 
 8 E0024_CNA -0.294 -0.302 
 9 E0026_CNA -0.302 -0.152 
10 E0028_CNA -0.173 -0.123 
# ℹ 46 more rows</code></pre>
</div>
</div>
<p>Then, we join this into the metadata, too.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>erie_meta <span class="ot">&lt;-</span> erie_meta <span class="sc">%&gt;%</span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(erie_nmds_tib)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Joining with `by = join_by(SampleID)`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>erie_meta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 56 × 37
   SampleID  Date       Station   Shore Fraction Type  Samplenum Date_year Month
   &lt;chr&gt;     &lt;date&gt;     &lt;chr&gt;     &lt;chr&gt; &lt;chr&gt;    &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;
 1 E0010_CNA 2014-06-16 nearshor… near… CNA      samp… E0010     6/16/14   June 
 2 E0012_CNA 2014-06-16 nearshor… near… CNA      samp… E0012     6/16/14   June 
 3 E0014_CNA 2014-06-16 offshore  offs… CNA      samp… E0014     6/16/14   June 
 4 E0016_CNA 2014-06-30 nearshor… near… CNA      samp… E0016     6/30/14   June 
 5 E0018_CNA 2014-06-30 nearshor… near… CNA      samp… E0018     6/30/14   June 
 6 E0020_CNA 2014-06-30 offshore  offs… CNA      samp… E0020     6/30/14   June 
 7 E0022_CNA 2014-07-08 nearshor… near… CNA      samp… E0022     7/8/14    July 
 8 E0024_CNA 2014-07-08 nearshor… near… CNA      samp… E0024     7/8/14    July 
 9 E0026_CNA 2014-07-08 offshore  offs… CNA      samp… E0026     7/8/14    July 
10 E0028_CNA 2014-07-14 nearshor… near… CNA      samp… E0028     7/14/14   July 
# ℹ 46 more rows
# ℹ 28 more variables: Days &lt;dbl&gt;, pH &lt;dbl&gt;, SpCond &lt;dbl&gt;, StationDepth &lt;dbl&gt;,
#   Secchi &lt;dbl&gt;, Temp &lt;dbl&gt;, Turbidity &lt;dbl&gt;, ParMC &lt;dbl&gt;, DissMC &lt;dbl&gt;,
#   Phycocyanin &lt;dbl&gt;, Chla &lt;dbl&gt;, DOC &lt;dbl&gt;, SRP &lt;dbl&gt;, TP &lt;dbl&gt;, TDP &lt;dbl&gt;,
#   POC &lt;dbl&gt;, PON &lt;dbl&gt;, PP &lt;dbl&gt;, N.P &lt;dbl&gt;, Nitrate &lt;dbl&gt;, Ammonia &lt;dbl&gt;,
#   H2O2 &lt;dbl&gt;, CDOM &lt;dbl&gt;, sequencingDepth &lt;int&gt;, pcoa.1 &lt;dbl&gt;, pcoa.2 &lt;dbl&gt;,
#   nmds.1 &lt;dbl&gt;, nmds.2 &lt;dbl&gt;</code></pre>
</div>
</div>
<p>Let’s save all of our hard work.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>erie_meta <span class="sc">%&gt;%</span> <span class="fu">saveRDS</span>(<span class="st">"clean_data/erie_meta_ordination.rds"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="pcoa-and-nmds-plots" class="level2">
<h2 class="anchored" data-anchor-id="pcoa-and-nmds-plots">PCoA and NMDS plots</h2>
<p>Before we do any plotting we will turn the month variable into a factor. We do this simply for the purpose of plotting. By default, ggplot will order character variables alphabetically, so that the months would be ordered August, July, June, October, September in plot legends. In contrast, we can specify the order of factor variables however we like.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>erie_meta <span class="ot">&lt;-</span> erie_meta <span class="sc">%&gt;%</span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Month =</span> </span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>           <span class="fu">factor</span>(Month, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"June"</span>, <span class="st">"July"</span>, <span class="st">"August"</span>, <span class="st">"September"</span>, <span class="st">"October"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let’s create our PCoA plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> erie_meta, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> pcoa<span class="fl">.1</span>, <span class="at">y =</span> pcoa<span class="fl">.2</span>, <span class="at">color =</span> Month, <span class="at">shape =</span> Station)) <span class="sc">+</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#a65628"</span>, <span class="st">"red"</span>, <span class="st">"#ffae19"</span>,</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"#4daf4a"</span>, <span class="st">"#1919ff"</span>, <span class="st">"darkorchid3"</span>, <span class="st">"magenta"</span>)</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> Month), <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">colour =</span> <span class="st">"grey90"</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="060_Erie_Data_Unconstrained_Ordination_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid" width="672"></p>
</div>
</div>
<p>And now we plot create the NMDS plot.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> erie_meta, <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x =</span> nmds<span class="fl">.1</span>, <span class="at">y =</span> nmds<span class="fl">.2</span>, <span class="at">color =</span> Month, <span class="at">shape =</span> Station)) <span class="sc">+</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#a65628"</span>, <span class="st">"red"</span>, <span class="st">"#ffae19"</span>,</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>    <span class="st">"#4daf4a"</span>, <span class="st">"#1919ff"</span>, <span class="st">"darkorchid3"</span>, <span class="st">"magenta"</span>)</span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> Month), <span class="at">alpha =</span> <span class="fl">0.7</span>, <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">colour =</span> <span class="st">"grey90"</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="060_Erie_Data_Unconstrained_Ordination_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid" width="672"></p>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>